osquery/tables/system/windows/shared_resources.cpp

```diff
@@ -8,45 +8,102 @@
+const std::string kWin32ShareQuery{"SELECT * FROM Win32_Share"};
+
+const std::unordered_map<long, std::string> kShareTypeNameMap = {
+    {0, "Disk Drive"},
+    {1, "Print Queue"},
+    {2, "Device"},
+    {3, "IPC"},
+    {2147483648, "Disk Drive Admin"},
+    {2147483649, "Print Queue Admin"},
+    {2147483650, "Device Admin"},
+    {2147483651, "IPC Admin"}};
+
+const std::string& getShareTypeName(const long& share_type) {
+  static const std::string kInvalidShareTypeName;
+
+  auto it = kShareTypeNameMap.find(share_type);
+  if (it == kShareTypeNameMap.end()) {
+    return kInvalidShareTypeName;
+  }
+
+  return it->second;
+}
+
+} // namespace
 
 QueryData genShares(QueryContext& context) {
-  QueryData results_data;
-
-  const auto request =
-      WmiRequest::CreateWmiRequest("SELECT * FROM Win32_Share");
-  if (request && request->getStatus().ok()) {
-    const std::vector<WmiResultItem>& results = request->results();
-    for (const auto& result : results) {
-      Row r;
-      long lPlaceHolder;
-      bool bPlaceHolder;
-
-      result.GetString("Description", r["description"]);
-      result.GetString("InstallDate", r["install_date"]);
-      result.GetString("Status", r["status"]);
-      auto status = result.GetBool("AllowMaximum", bPlaceHolder);
-      r["allow_maximum"] = status ? INTEGER(bPlaceHolder) : "-1";
-      status = result.GetLong("MaximumAllowed", lPlaceHolder);
-      r["maximum_allowed"] = status ? INTEGER(lPlaceHolder) : "-1";
-      result.GetString("Name", r["name"]);
-      result.GetString("Path", r["path"]);
-      status = result.GetLong("Type", lPlaceHolder);
-      r["type"] = status ? INTEGER(lPlaceHolder) : "-1";
-
-      results_data.push_back(r);
+  auto exp_request = WmiRequest::CreateWmiRequest(kWin32ShareQuery);
+  if (exp_request.isError()) {
+    const auto& error = exp_request.getError();
+
+    LOG(ERROR) << "The following WMI query could not be constructed: "
+               << kWin32ShareQuery << ". " << error.getMessage();
+
+    return {};
+  }
+
+  auto request = exp_request.take();
+  if (!request.getStatus().ok()) {
+    LOG(ERROR) << "The following WMI query could not be executed correctly: "
+               << kWin32ShareQuery;
+    return {};
+  }
+
+  const auto& wmi_item_list = request.results();
+  if (wmi_item_list.empty()) {
+    return {};
+  }
+
+  QueryData row_list;
+
+  for (const auto& wmi_item : wmi_item_list) {
+    Row row;
+    wmi_item.GetString("Description", row["description"]);
+    wmi_item.GetString("InstallDate", row["install_date"]);
+    wmi_item.GetString("Status", row["status"]);
+    wmi_item.GetString("Name", row["name"]);
+    wmi_item.GetString("Path", row["path"]);
+
+    bool allow_maximum{};
+    auto status = wmi_item.GetBool("AllowMaximum", allow_maximum);
+    row["allow_maximum"] = INTEGER(status.ok() ? allow_maximum : -1);
+
+    long raw_maximum_allowed_value{};
+    status = wmi_item.GetLong("MaximumAllowed", raw_maximum_allowed_value);
```

Comment:
I'm fine either way. But does it even make sense to populate this field if `AllowMaximum` is set?

Reply:
I was about to go down the same route, but I noticed that this field is always populated and retrieved correctly regardless of what `AllowMaximum` has been set to